contracts/helpers/MultiBlockHarvester.sol

```diff
@@ -0,0 +1,213 @@
+        agToken.burnSelf(amountOut, address(this));
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    function _rebalance(
+        uint8 typeAction,
+        address yieldBearingAsset,
+        YieldBearingParams memory yieldBearingInfo,
+        uint256 amount
+    ) internal {
+        if (amount > maxMintAmount) revert TooBigAmountIn();
+        agToken.mint(address(this), amount);
+        _adjustAllowance(address(agToken), address(transmuter), amount);
+        if (typeAction == 1) {
+            address depositAddress = yieldBearingToDepositAddress[yieldBearingAsset];
+
+            if (yieldBearingAsset == XEVT) {
+                uint256 amountOut = transmuter.swapExactInput(
+                    amount,
+                    0,
+                    address(agToken),
+                    yieldBearingInfo.stablecoin,
+                    address(this),
+                    block.timestamp
+                );
+                _checkSlippage(amount, amountOut, yieldBearingInfo.stablecoin, depositAddress, false);
+                _adjustAllowance(yieldBearingInfo.stablecoin, address(depositAddress), amountOut);
+                (uint256 shares, ) = IPool(depositAddress).deposit(amountOut, address(this));
+                _adjustAllowance(yieldBearingAsset, address(transmuter), shares);
+                amountOut = transmuter.swapExactInput(
+                    shares,
+                    0,
+                    yieldBearingAsset,
+                    address(agToken),
+                    address(this),
+                    block.timestamp
+                );
+                agToken.burnSelf(amountOut, address(this));
+            } else if (yieldBearingAsset == USDM) {
+                uint256 amountOut = transmuter.swapExactInput(
+                    amount,
+                    0,
+                    address(agToken),
+                    yieldBearingInfo.stablecoin,
+                    address(this),
+                    block.timestamp
+                );
+                _checkSlippage(amount, amountOut, yieldBearingInfo.stablecoin, depositAddress, false);
+                IERC20(yieldBearingInfo.stablecoin).safeTransfer(depositAddress, amountOut);
+            }
+        } else {
+            uint256 amountOut = transmuter.swapExactInput(
+                amount,
+                0,
+                address(agToken),
+                yieldBearingAsset,
+                address(this),
+                block.timestamp
+            );
+            address depositAddress = yieldBearingToDepositAddress[yieldBearingAsset];
+            _checkSlippage(amount, amountOut, yieldBearingAsset, depositAddress, false);
```

Comment:
Same here, some redundancy, `address depositAddress` can be moved above `if (typeAction == 1)`